src/controllers/inventoryController.spec.js

```diff
@@ -0,0 +1,260 @@
+        project: mongoose.Types.ObjectId(mockReq.params.projectId),
+        wbs: mongoose.Types.ObjectId(mockReq.params.wbsId),
+        $gte: { quantity: 0 },
+      });
+
+      assertResMock(404, new Error('an error occured'), response, mockRes);
+    });
+    test('Ensure getAllInvInProjectWBS Returns status 200 if results are found sorted and popluated', async () => {
+      const { getAllInvInProjectWBS } = makeSut();
+      const hasPermissionSpy = mockHasPermission(true);
+
+      const findObject = {
+        populate: () => {},
+        inventory: [{ project: 'someName', wbs: null }],
+      };
+
+      const sortedObject = {
+        sort: () => {},
+      };
+
+      const findSpy = jest.spyOn(inventoryItem, 'find').mockImplementationOnce(() => {
+        return findObject;
+      });
+      jest.spyOn(findObject, 'populate').mockImplementationOnce(() => {
+        return findObject;
+      });
+      jest.spyOn(findObject, 'populate').mockImplementationOnce(() => {
+        return findObject;
+      });
+      jest.spyOn(findObject, 'populate').mockImplementationOnce(() => {
+        return sortedObject;
+      });
+      jest.spyOn(sortedObject, 'sort').mockImplementationOnce(() => {
+        return Promise.resolve(findObject.inventory);
+      });
+
+      const response = await getAllInvInProjectWBS(mockReq, mockRes);
+
+      expect(findSpy).toHaveBeenCalledWith({
+        project: mongoose.Types.ObjectId(mockReq.params.projectId),
+        wbs: mongoose.Types.ObjectId(mockReq.params.wbsId),
+        $gte: { quantity: 0 },
+      });
+
+      assertResMock(200, findObject.inventory, response, mockRes);
+    });
+  });
+
+  describe('postInvInProjectWBS', () => {
+    test("Ensure postInvInProjectWBS Returns error 403 if the user doesn't have the postInvInProjectWBS permission", async () => {
+      const { postInvInProjectWBS } = makeSut();
+      const hasPermissionSpy = mockHasPermission(false);
+      const response = await postInvInProjectWBS(mockReq, mockRes);
+
+      expect(hasPermissionSpy).toHaveBeenCalledWith(mockReq.body.requestor, 'postInvInProjectWBS');
+      assertResMock(403, 'You are not authorized to view inventory data.', response, mockRes);
+    });
+    test('Ensure postInvInProjectWBS Returns error 400 if valid project, but quantity and id are necessary as well as valid wbs if sent in and not Unassigned', async () => {
+      const { postInvInProjectWBS } = makeSut();
+      const hasPermissionSpy = mockHasPermission(true);
+      const findObject = {
+        populate: () => {},
+        inventory: [{ project: 'someName', wbs: null }],
+      };
+      mockReq.body.quantity = 1;
+      mockReq.body.typeId = '';
+
+      const projectExists = {
+        select: () => {},
+        lean: () => {},
+      };
+
+      const wbsExists = {
+        select: () => {},
+        lean: () => {},
+      };
+
+      const findSpyProjects = jest.spyOn(projects, 'findOne').mockImplementationOnce(() => {
```

Comment:
Make sure you assert that these spy methods are called with the correct arguments as well.